### PR TITLE
feat: support Dovecot DIGEST-MD5

### PR DESCRIPTION
**Pull request by bestlong**

### CODE REVIEW

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -44,12 +44,21 @@ jobs:
     steps:
     - uses: actions/checkout@v2
 
+    - name: Install Dovecot
+      run: |
+        sudo apt-get update -q
+        sudo apt-get install -yq dovecot-core
+        doveconf
+        cat /etc/passwd
```
Comment (DavidGoodwin) [inline .github/workflows/php.yml:64]: can you drop the cat / sudo lines please?

File: functions.inc.php [hunk 2]
```diff
@@ -1076,11 +1077,14 @@ function _pacrypt_dovecot($pw, $pw_db = '')
...
 
-    # digest-md5 hashes include the username - until someone implements it, let's declare it as unsupported
+    $doveadm_options = '';
+
     if (strtolower($method) == 'digest-md5') {
-        throw new Exception("Sorry, \$CONF['encrypt'] = 'dovecot:digest-md5' is not supported by PostfixAdmin.");
+        if (empty($username)) {
+            throw new Exception("\$CONF['encrypt'] = 'dovecot:digest-md5' require username.");
+        }
+        $doveadm_options = ' -u ' . $username;
```
Comment (DavidGoodwin) [inline functions.inc.php:1086]: $username needs to pass through escapeshellarg?

File: functions.inc.php [hunk 3]
```diff
@@ -1105,7 +1109,7 @@ function _pacrypt_dovecot($pw, $pw_db = '')
 
     $pipes = [];
 
-    $pipe = proc_open("$dovecotpw '-s' $method$dovepasstest", $spec, $pipes);
+    $pipe = proc_open("$dovecotpw '-s' $method$dovepasstest$doveadm_options", $spec, $pipes);
```
Comment (DavidGoodwin) [inline functions.inc.php:1112]: i think this needs rewriting to be a little clearer - e.g. 

.... {$method}{$dovepasstest}{$doveadm_options} ?

I wonder why -s has quote marks around it?

File: functions.inc.php [hunk 4]
```diff
@@ -1323,9 +1327,10 @@ function _php_crypt_random_string($characters, $length)
  *
  * @param string $pw
  * @param string $pw_db optional encrypted password
+ * @param string $username optional
```
Comment (DavidGoodwin) [inline functions.inc.php:1330]: can you add something to the username comment to say it's currently only relevant for dovecot DIGEST-MD5 format

**Comment by DavidGoodwin** [comment]
Thank you for taking the time to contribute to PostfixAdmin.

File: .github/workflows/php.yml (hunk 1 above)
Comment (bestlong) [reply .github/workflows/php.yml]: It will clean up eventually, but sudo installation of dovecot is necessary.

File: functions.inc.php (hunk 2 above)
Comment (bestlong) [reply functions.inc.php]: fix

**Review by DavidGoodwin — APPROVED** [review]
LGTM, thank you

**Pull request merged**